watershed_workflow/split_hucs.py

```diff
@@ -174,7 +175,18 @@ def polygon(self, i):
                 segs.append(self.segments[s])
 
         ml = shapely.ops.linemerge(segs)
-        assert (type(ml) is shapely.geometry.LineString)
+
+        if type(ml) is shapely.geometry.MultiLineString:
+            logging.debug("the lines may not be contiguous, try to make a new simple\
+                line using the line coordinates.")
+            # Put the sub-line coordinates into a list of sublists
+            coords = [list(line.coords) for line in ml]
```

Comment:
This seems a bit fragile.  How do you know the lines are in the right order?  Can you give an example where this breaks so I can see what is going wrong?